app/models/repo.rb

```diff
@@ -173,37 +159,12 @@ def self.order_by_subscribers
       order("count(users.id) DESC")
   end
 
-  def populate_issue(options = {})
-    page     = options[:page]  || 1
-    state    = options[:state] || "open"
-    response = GitHubBub.get(api_issues_path, state:     state,
-                                              page:      page,
-                                              sort:      'comments',
-                                              direction: 'desc')
-    response.json_body.each do |issue_hash|
-      logger.info "Issue: number: #{issue_hash['number']}, updated_at: #{issue_hash['updated_at']}"
-      Issue.find_or_create_from_hash!(issue_hash, self)
-    end
-    response
-  end
-
-
-  def populate_multi_issues!(options = {})
-    options[:state] ||= "open"
-    options[:page]  ||= 1
-    response = populate_issue(options)
-    until response.last_page?
-      options[:page] += 1
-      response = populate_issue(options)
-    end
-  end
-
   def update_from_github
-    resp = GitHubBub.get(repo_path).json_body
-    self.language    = resp['language']
-    self.description = resp['description']
-    self.stars_count = resp['stargazers_count']
-    self.save
+    self.update(
+      language: fetcher.json['language'],
```

Comment:
It looks like our Fetcher's contract is to always return a "null object" of the same type that it would have received. e.g. `json` here should return an empty hash on request failure.